W 11932809 test and debug cv (#91)
mulesoft/docs-rpa: rpa-builder/6.0/modules/ROOT/pages/analyzing-problems-with-analysis-packages.adoc

= Analyzing Problems with Analysis Packages

> After the business process runs locally in RPA Builder, you can upload it to RPA Manager, move it to the test phase, run it on an RPA Bot, and start a long-term test. We recommend a test period of three days before using the business process productively. If menu runs contain errors, RPA Manager generates Analysis Packages that you can download.

Reviewer: ```suggestion
After the business process runs locally in RPA Builder, you can upload it to RPA Manager, move it to the test phase, run it on an RPA Bot, and start a long-term test. Run tests for at least three days before using the business process productively. If menu runs contain errors, RPA Manager generates Analysis Packages that you can download.
```

Reviewer: this seems like a nit to me, but I think the style guide says not use constructions like "We recommend..."

> The Analysis Package provides detailed information on the run in the same format as if you had executed the Workflow locally and also enables you to fix the error. Elements, which have caused an error or a warning, are marked with an expandable red or yellow frame respectively. This frame contains more detailed information concerning the run results. Use the Analysis Packages to understand and fix errors for each failed Workflow run. If you see the same error repeatedly, fix the error once and start collecting results again.

Reviewer: ```suggestion
The Analysis Package provides detailed information on the run in the same format as if you had executed the Workflow locally. Use this information to fix any errors. Elements, which have caused an error or a warning, are marked with an expandable red or yellow frame. This frame contains more detailed information concerning the run results. Use the Analysis Packages to understand and fix errors for each failed Workflow run. If you see the same error repeatedly, fix the error once and start collecting results again.
```

> == Download and Analysis Package

Reviewer: there should be an intro sentence between the heading and command

+
If you don't have any test plans available, click *Create new test plan* and follow the instructions to create one.
. Click *Update* to refresh the test run results.
. Click the result number in the *Failure* column.
+
image:test-plan-error.png[The Test Results window showing a Failure]
+
If there are no errors, the *Failure* column shows `0` results and there's no Analysis Package available to download.
. Click the *Download* button.
+
image:download-analysis-package.png[The Process Run Results Window and the Download analysis package button]

== Load and Analysis Package

To load an Analysis Package:

. In RPA Builder, open any Workflow to make the Workbench menu appear.
. In the Workbench menu, click *Tools*.
. Click *Load Analysis Package* and select the location of the Analysis Package file.
+
image:workbench-tools-menu.png[The Workbench Tools Menu, 30%, 30%]

== Analysis Package Directory

RPA Builder creates the `custom` directory after you start the menu locally for the first time, either with the *Run Workflow* button or by pressing _F7_. The path to the `custom` directory is: `C:\Users\[UserName]\Documents\MuleSoft\Cache\[Process-GUID]\[Activity- GUID]\Robot\custom\`.

> You can right-click on the Workflow tab and select the *Open Workflow Location* option, which automatically redirects you to the `C:\Users\[UserName]\Documents\MuleSoft\Cache\[Process-GUID]\[Activity- GUID]\` folder.

Reviewer: ```suggestion
Right-click on the Workflow tab and select the *Open Workflow Location* option, which automatically redirects you to the `C:\Users\[UserName]\Documents\MuleSoft\Cache\[Process-GUID]\[Activity- GUID]\` folder.
```

> You can add extra data like screenshots or log files to the analysis package by saving them to the project’s `custom` folder. RPA Manager packs all files from this folder into the analysis package in case of an error.

Reviewer: How come analysis package isn't capitalized here?

. Click the result number in the *Failure* column.
+
image:test-plan-error.png[The Test Results window showing a Failure]
+
If there are no errors, the *Failure* column shows `0` results and there's no Analysis Package available to download.
. Click the *Download* button.
+
image:download-analysis-package.png[The Process Run Results Window and the Download analysis package button]
////
If an error has occurred, it shows in the *Failure* column. If you click on the result, you get to the *Process Run Results* window. You can now download the Analysis Package here by clicking on the *Download* icon.
////

== Load and Analysis Package

To load an Analysis Package:

. In RPA Builder, open any Workflow to make the Workbench menu appear.
. In the Workbench menu, click *Tools*.
. Click *Load Analysis Package* and select the location of the Analysis Package file.
+
image:workbench-tools-menu.png[The Workbench Tools Menu, 30%, 30%]

//TODO: create anchor load-analysis-package in PR#84 and merge it before merging this content.
//You can then load the Analysis Package into RPA Builder via the Workbench menu item *Tools* -> *Load Analysis Package*.

== Analysis Package Directory

> RPA Builder creates the `custom` directory after you start the menu locally for the first time, either with the *Run Workflow* button or by pressing _F7_. The path to the `custom` directory is: `C:\Users\[UserName]\Documents\MuleSoft\Cache\[Process-GUID]\[Activity- GUID]\Robot\custom\`. You can also right-click on the Workflow tab and select the *Open Workflow Location* option, which automatically redirects you to the correct directory.

Reviewer 2: The menu item only leads to the project folder. From there the user has to enter the Robot and then the custom folder.

To be correct, the sample folder looks like this:
C:\Users\[UserName]\Documents\MuleSoft\Cache\[Process-ID]\[Activity- GUID]
